File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <queue>
#include <algorithm>
using namespace std;

int N, M, V;
vector<vector<int>> graph(1001, vector<int>());
bool visit[1001];

void DFS(int cur) {
    cout << cur << " ";
    visit[cur] = 1;

    for (int i = 0; i < graph[cur].size(); i++) {
        if (!visit[graph[cur][i]]) DFS(graph[cur][i]);
    }
}

void BFS() {
    queue<int> q;
    q.push(V);
    visit[V] = 1;

    while (!q.empty()) {
        int cur = q.front();
        q.pop();
        cout << cur << " ";
        for (int i = 0; i < graph[cur].size(); i++) {
            if (!visit[graph[cur][i]]) {
                q.push(graph[cur][i]);
                visit[graph[cur][i]] = 1;
            }
        }
    }
}

int main() {
    cin >> N >> M >> V;

    int x, y;
    for (int i = 0; i < M; i++) {
        cin >> x >> y;
        graph[x].push_back(y);
        graph[y].push_back(x);
    }

    for (int i = 0; i <= N; i++) {
        visit[i] = 0;
        sort(graph[i].begin(), graph[i].end());
    }
    DFS(V);
    cout << "\n";
    for (int i = 0; i <= N; i++) {
        visit[i] = 0;
    }
    BFS();


}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;


class Main {
    public static int N, M, V;

    public static List<List<Integer>> list;
    public static boolean[] visit;

    public static void dfs(int cur) {
        System.out.print(cur + " ");
        visit[cur] = true;

        for (int next : list.get(cur)) {
            if (!visit[next]) dfs(next);
        }

    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());

        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());

        list = new ArrayList<>(N+1);
        visit = new boolean[N+1];

        for (int i = 0; i <= N; i++) {
            list.add(new ArrayList<>());
        }

        for (int i = 0; i < M; i++) {
            st = new StringTokenizer(br.readLine());
            int n1 = Integer.parseInt(st.nextToken());
            int n2 = Integer.parseInt(st.nextToken());
            list.get(n1).add(n2);
            list.get(n2).add(n1);
        }

        for(List<Integer> l : list) {
           Collections.sort(l);
        }

        dfs(V);
        System.out.println();

        Arrays.fill(visit, false);

        Queue<Integer> que = new LinkedList<>();
        que.offer(V);
        visit[V] = true;

        while(!que.isEmpty()) {
            int cur = que.poll();
            System.out.print(cur + " ");

            for (int next : list.get(cur)) {
                if (next!=0 && !visit[next]) {
                    que.offer(next);
                    visit[next] = true;
                }
            }
        }

    }
}